CEP170B.py

```diff
@@ -0,0 +1,139 @@
+import os
+from abc import ABC, abstractmethod
+from Bio import SeqIO
+from Bio.SeqUtils import gc_fraction
+from typing import Tuple
+
+
+def filter_fastq(input_path: str, 
+                 output_filename: str = None, 
+                 gc_bounds: Tuple[int, int] = (0, 100), 
+                 length_bounds: Tuple[int, int] = (0, 2**32), 
+                 quality_threshold: int = 0) -> None:
+    '''
+    Filter FASTQ-sequences based on entered requirements.
+    
+    Arguments:
+        - input_path (str): path to the file with FASTQ-sequences
+        - output_filename (str): name of the output file with 
+        filtered FASTQ-sequences
+        - gc_bounds (tuple or int, default = (0, 100)): GC-content
+        interval (percentage) for filtering. Tuple if contains 
+        lower and upper bounds, int if only contains an upper bound.
+        - length_bounds (tuple or int, default = (0, 2**32)): length 
+        interval for filtering. Tuple if contains lower and upper 
+        bounds, int if only contains an upper bound.
+        - quality_threshold (int, default = 0): threshold value of average 
+        read quality for filtering.
+
+    Note: the output file is saved to the /fastq_filtrator_results 
+    directory. The default output file name is the name of the input file.
+    '''
+    with open(input_path, "r") as handle:
+        records = [record for record in SeqIO.parse(handle, "fastq")]
+
+    filtered_records = []
+    for record in records:
```

Comment:
Здесь была возможность использовать SeqIO:
```suggestion
    for i, record in enumerate(SeqIO.parse(input_path, "fastq")):
```